documentation/design-docs/eventcounters.md

```diff
@@ -0,0 +1,306 @@
+# EventCounters
+
+## Introduction
+EventCounters are .NET Core APIs used for lightweight, cross-platform, and near real-time performance metric collection. EventCounters that were added to replace the "performance counters" on the .NET Framework. This documentation serves as a guide on what they are, how they are implemented, how to use them, and how to consume them. 
+
+EventCounters have existed for a while (and is actually supported in .NET Framework as well) but the API has been extended starting in .NET Core 3.0. In addition to that, the .NET Core runtime (CoreCLR) and few .NET libraries have started publishing basic diagnostics information using EventCounters starting in .NET Core 3.0. This document is written targetting versions of .NET Core 3.0 or later.
+
+This document serves to explain some of the concepts and design choices behind EventCounters API, how to use them, and how they can be consumed both in-proc and out-of-proc with some sample code.
+
+
+## Conceptual Overview
+EventCounters collect numeric values over some time and report them. The .NET Core runtime (CoreCLR) has several EventCounters, which collect and publish basic performance metrics related to CPU usage, memory usage, GC heap statistics, threads and locks statistics, etc. These serve as a basic performance guideline that can be easily consumed at a relatively cheap cost (that is, turning them on does not cause performance regression, so they can be used in production scenarios). 
+
+Apart from the EventCounters that are already provided by the .NET runtime or the rest of the framework (i.e. ASP.NET, gRPC, etc.), you may choose to implement your own EventCounters to keep track of various metrics for your service. 
+
+EventCounters live as a part of an [EventSource](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource?view=netcore-3.0). Like any other events on an `EventSource`, they can be consumed both in-proc and out-of-proc via [EventListener](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventlistener?view=netcore-3.0) and EventPipe/ETW.
+
+![EventCounter](EventCounters.jpg)
```

Comment:
For the image instead of having boxes with EventPipe and ETW I'd suggest we use a big arrow from left to right and label the arrow "EventPipe or ETW". This suggests that EventPipe is channel the data flows through as opposed to a particular intermediate destination the data arrives at and then leaves.

I would also split the PerfView/dotnet-trace box into two boxes. PerfView is a direct collector of ETW data and doesn't have to be used in conjunction with dotnet-trace.